python/vowpalwabbit/DFtoVWtoTensorboard.py

```diff
@@ -0,0 +1,90 @@
+import tensorboardX as tx
+
+
+class VWtoTensorboard:
+	"""The object of this class would be passed as a callback to DFtoVWtoTensorboard fit method to ensure writing of logs for Tensorboard"""
+
+	def __init__(self, logdir):
+		"""Construct a VWtoTensorboard object
+
+		Parameters
+		----------
+
+		logdir : str
+			A string specifying the log directory for Tensorboard logs
+
+		Returns
+		-------
+
+		self : VWtoTensorboard
+		"""
+		self.file_writer = tx.SummaryWriter(logdir)   # creating file writer
+		self.iteration = 0   # This would keep value of current iteration 
+
+	def emit_learning_metrics(self, average_loss, since_last):
+		"""This method for now logs the metrics given as arguments for Tensorboard visualization
+
+		Parameters
+		----------
+
+		average_loss : float
+				This is the average_loss value to be logged for Tensorboard
+		since_last   : float
+				This is the since_last value to be logged for Tensorboard
+
+		Returns
+		-------
+
+		None
+		"""
+		self.file_writer.add_scalar('average_loss', average_loss, self.iteration)  # logging average_loss on each iteration
+		self.file_writer.add_scalar('since_last', since_last, self.iteration)   # logging since_last on each iteration
+		self.iteration += 1  # Now increment this as the incremented value is for next iteration
+
+
+
+class DFtoVWtoTensorboard:
+	"""This class provides capability for learning possible on DFtoVW object, provides capability to 
+	print metrics (average_loss, since_last, label, prediction, num_features) and also provides support 
+	to log these metrics for Tensorboard Visualization"""
+
+	def __init__(self, df_to_vw, vw):
+		"""Construct a DFtoVWtoTensorboard object
+
+		Parameters
+		----------
+
+		df_to_vw : A DFtoVW object
+				This would be used to construct examples 
+		vw		 : A vowpalwabbit object
+ 				This would be used for learning using the algorithm specified in the object, and also to compute a few metrics
+
+		Returns
+		-------
+
+		self : DFtoVWtoTensorboard
+		"""
+		self.vw = vw  # As this needs to be remembered
+		self.vw_formatted_data = df_to_vw.convert_df()
```

Comment:
This part introduce a tight coupling with `DFtoVW` class that is not necessary IMO. 

The examples could be the result of `DFtoVW.convert_df()` **but could well have been directly built by the user**. It seems better to just pass a list of string `vw_examples` as argument instead of a `DFtoVW` object.

Reply:
I will look into that